pkg/serial/handshake.go

```diff
@@ -40,6 +40,9 @@ const (
 
 // PurgeIncoming is used to clear a channel of bytes prior to handshaking
 func PurgeIncoming(conn net.Conn) {
+	if tracing {
```

Comment:
But that would either require passing a new param to Begin, or doing more calls to trace hence breaking that defer End(Begin(msg)) chain